theforeman.org/pipelines/test/rpm_copr_packaging.groovy

```diff
@@ -0,0 +1,170 @@
+                              continue
+                            }
+
+                            sh(script: "git checkout origin/${env.ghprbTargetBranch}", label: "git checkout target_branch")
+                            old_version = query_rpmspec(old_spec_path, '%{VERSION}')
+                            old_release = query_rpmspec(old_spec_path, '%{RELEASE}')
+
+                            sh(script: "git checkout -", label: "git checkout source_branch")
+                            new_version = query_rpmspec(new_spec_path, '%{VERSION}')
+                            new_release = query_rpmspec(new_spec_path, '%{RELEASE}')
+
+                            compare_version = sh(
+                              script: "rpmdev-vercmp ${old_version} ${new_version}",
+                              returnStatus: true,
+                              label: "rpmdev-vercmp"
+                            )
+
+                            compare_release = sh(
+                              script: "rpmdev-vercmp ${old_release} ${new_release}",
+                              returnStatus: true,
+                              label: "rpmdev-vercmp"
+                            )
+
+                            compare_new_to_one = sh(
+                              script: "rpmdev-vercmp 1 ${new_release}",
+                              returnStatus: true,
+                              label: "rpmdev-vercmp"
+                            )
+
+                            if (compare_version != VERCMP_EQUAL && (compare_new_to_one == VERCMP_OLDER || compare_new_to_one == VERCMP_EQUAL)) {
+                                echo "New version and release is reset to 1 for ${package_name}"
+                            } else if (compare_version != VERCMP_EQUAL && compare_new_to_one == VERCMP_NEWER) {
+                                // new version, but release was not reset
+                                sh """
+                                    echo 'Version updated but release was not reset back to 1 for ${package_name}"'
+                                    exit 1
+                                """
+                            } else if (compare_version == VERCMP_EQUAL && compare_release == VERCMP_NEWER) {
+                                echo "Version remained the same and release is reset to 1 for ${package_name}"
+                            } else {
+                                sh """
+                                    echo 'Version or release needs updating for ${package_name}"'
+                                    exit 1
+                                """
+                            }
+                        }
+                    }
+                }
+            }
+        }
+
+        stage('Scratch Build Packages') {
+            when {
+                expression { packages_to_build }
+            }
+            steps {
+
+                withCredentials([file(credentialsId: 'theforeman-bot-copr', variable: 'copr_config')]) {
```

Comment:
@evgeni You asked me if we could do authentication different than we do Koji. This is my solution, to keep a secrets file that is a copy of the API configuration file a user gets from Copr in Jenkins credentials and deploy it during the calls that need it so it gets obfuscated and cleaned up automatically.